pkg/quota/evaluator/core/persistent_volume_claims.go

```diff
@@ -51,68 +54,127 @@ func listPersistentVolumeClaimsByNamespaceFuncUsingClient(kubeClient clientset.I
 	}
 }
 
+// ResourceByStorageClass returns a quota resource name by storage class.
+func ResourceByStorageClass(storageClass string, resourceName api.ResourceName) api.ResourceName {
+	return api.ResourceName(string(storageClass + storageClassSuffix + string(resourceName)))
+}
+
+// V1ResourceByStorageClass returns a quota resource name by storage class.
+func V1ResourceByStorageClass(storageClass string, resourceName v1.ResourceName) v1.ResourceName {
+	return v1.ResourceName(string(storageClass + storageClassSuffix + string(resourceName)))
+}
+
 // NewPersistentVolumeClaimEvaluator returns an evaluator that can evaluate persistent volume claims
 // if the specified shared informer factory is not nil, evaluator may use it to support listing functions.
 func NewPersistentVolumeClaimEvaluator(kubeClient clientset.Interface, f informers.SharedInformerFactory) quota.Evaluator {
-	allResources := []api.ResourceName{api.ResourcePersistentVolumeClaims, api.ResourceRequestsStorage}
+	fixedResources := []api.ResourceName{api.ResourcePersistentVolumeClaims, api.ResourceRequestsStorage}
+	usageFunc := PersistentVolumeClaimUsageFunc
+	constraintsFunc := makePersistentVolumeClaimConstraintsFunc(usageFunc)
 	listFuncByNamespace := listPersistentVolumeClaimsByNamespaceFuncUsingClient(kubeClient)
 	if f != nil {
 		listFuncByNamespace = generic.ListResourceUsingInformerFunc(f, schema.GroupResource{Resource: "persistentvolumeclaims"})
 	}
-
 	return &generic.GenericEvaluator{
```

Comment:
> I agree this has morphed into something more like registry.Store since we now pass in functions instead of static values that are used to implement said functions. I was trying to use the StaticMatchedResourceNamesFunc as a reasonable first-step at this point.

I think I'd rather end up with two cases: 
 1. plain counts: these should be generic and not require any code
 2. other quota: these should implement the interface.

This pull doesn't have to solve those problems, but I don't think it should take us further away.  Rather than change `GenericEvaluator` to suit this special quota evaluator, I think this should just implement the interface itself.